curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/PersistentEphemeralNode.java

```diff
@@ -233,8 +236,15 @@ else if ( event.getResultCode() == KeeperException.Code.OK.intValue() )
                 {
                     path = event.getName();
                 }
+                else if ( event.getResultCode() == KeeperException.Code.NOAUTH.intValue() )
+                {
+                	log.warn("Client does not have authorisation to write ephemeral node at path {}", path);
```

Comment:
Maybe we should just log an error and stop processing for any code that is not in RetryLoop.shouldRetry()? I was just trying to minimise the impact of the change as it was previously just handling the NodeExists and OK cases and retrying everything else.